rls/src/cmd.rs

```diff
@@ -172,15 +172,55 @@ pub fn run() {
     }
 }
 
-fn def(file_name: &str, row: &str, col: &str) -> Request<requests::Definition> {
+macro_rules! read_unsigned {
```

Comment:
Hm, that's a lot of machinery that's only used for `def`, which isn't ideal... maybe going for a [`scanf-like`](https://github.com/oli-obk/rust-si) approach seems like a better idea? If not, I'd go for a more principled approach here - either trying to unify the macros interface here somehow or by applying the same approach to every subcommand here

Reply:
> maybe going for a [`scanf-like`](https://github.com/oli-obk/rust-si) approach seems like a better idea?

@Xanewok I was actually using macros in the first place to try not to repeat too much code while removing the `panic`s since I was advised not to introduce new dependencies here (https://github.com/rust-lang/rls/issues/1472#issuecomment-498224681). Closing then to evaluate other alternatives.